spec.bs

```diff
@@ -934,20 +934,26 @@ To <dfn>validate and convert auction ad config</dfn> given an {{AuctionAdConfig}
       1. Set |auctionConfig|'s [=auction config/seller signals=] to the result of
         [=serializing a JavaScript value to a JSON string=], given |result|.
     * To handle an error, set |auctionConfig|'s [=auction config/seller signals=] to failure.
-1. If |config|["{{AuctionAdConfig/directFromSellerSignals}}"] [=map/exists=]:
-  1. TODO: The receiving end of this isn't specified yet, so there is no place to put the computed value.
+1. If |config|["{{AuctionAdConfig/directFromSellerSignalsHeaderAdSlot}}"] [=map/exists=]:
   1. [=Handle an input promise in configuration=] given |auctionConfig| and
-    |config|["{{AuctionAdConfig/directFromSellerSignals}}"]:
+    |config|["{{AuctionAdConfig/directFromSellerSignalsHeaderAdSlot}}"]:
+    
+    Note: The JavaScript code calling {{Navigator/runAdAuction()}} is responsible for not resolving
+      |config|["{{AuctionAdConfig/directFromSellerSignalsHeaderAdSlot}}"] until a
+      {{WindowOrWorkerGlobalScope/fetch()}} call with the `{adAuctionHeaders: true}` option returns
+      a response, or alternatively, not calling {{Navigator/runAdAuction()}} until the
+      {{WindowOrWorkerGlobalScope/fetch()}} call returns a response.
+      TODO: Link to {{WindowOrWorkerGlobalScope/fetch()}} once the fetch patch PR is merged.
+
     * To parse the value |result|:
-      1. Let |directFromSellerSignalsPrefix| be the result of running the [=URL parser=] on |result|.
-      1. [=exception/Throw=] a {{TypeError}} if any of the following conditions hold:
-        * |directFromSellerSignalsPrefix| is failure;
-        * |directFromSellerSignalsPrefix| is not [=same origin=] with |auctionConfig|'s
-          [=auction config/seller=];
-        * |directFromSellerSignalsPrefix|'s [=url/query=] is not null.
-      1. [=Assert=]: |directFromSellerSignalsPrefix|'s [=url/scheme=] is "`https`".
+      1. Set |auctionConfig|'s [=auction config/direct from seller signals header ad slot=] to
+        |result|.
+      1. Set |auctionConfig|'s
+        [=auction config/config idl=]["{{AuctionAdConfig/directFromSellerSignalsHeaderAdSlot}}"] to
+        |result|.

```

Comment:
This seems broken because `{{AuctionAdConfig/directFromSellerSignalsHeaderAdSlot}}` is a promise, not a string. And this is precisely the promise that just resolved anyways, right?

Reply:
oh you're right. Then some existing similar cases like [auctionSignals](https://github.com/WICG/turtledove/blob/main/spec.bs#L922) have the same problem. Will fix those in another PR.

Reply:
https://github.com/WICG/turtledove/pull/854